src/plugins/ini/testmod_ini.c

```diff
@@ -103,7 +103,49 @@ static void test_plainIniWrite(char *fileName)
 	PLUGIN_CLOSE ();
 }
 
+static void test_plainIniEmptyWrite(char *fileName)
+{
+	Key *parentKey = keyNew ("user/tests/ini-write", KEY_VALUE,
+			elektraFilename(), KEY_END);
+	KeySet *conf = ksNew(0, KS_END);
+	PLUGIN_OPEN("ini");
+
+	KeySet *ks = ksNew (30,
+			keyNew ("user/tests/ini-write/nosectionkey",
+					KEY_VALUE, "nosectionvalue",
+					KEY_END),
+			keyNew ("user/tests/ini-write/section1",
+					KEY_BINARY,
+					KEY_END),
+			keyNew ("user/tests/ini-write/section1/key1",
+					KEY_VALUE, "value1",
+					KEY_END),
+			keyNew ("user/tests/ini-write/section1/key2",
+					KEY_VALUE, "value2",
+					KEY_END),
+			keyNew ("user/tests/ini-write/section2",
+					KEY_BINARY,
+					KEY_END),
+			keyNew ("user/tests/ini-write/section2/key3",
+					KEY_VALUE, "value3",
+					KEY_END),
+			keyNew ("user/tests/ini-write/section2/emptykey", KEY_META, "ini/empty", "", KEY_END),
+			KS_END);
+
+	succeed_if(plugin->kdbSet (plugin, ks, parentKey) >= 1,
```

Comment:
could you test full round-trip? also get it again and also check if keyset is equal.